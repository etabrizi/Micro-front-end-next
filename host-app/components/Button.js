import React from 'react';

const Button = () => <button>Next JS Button</button>;

export default Button;
