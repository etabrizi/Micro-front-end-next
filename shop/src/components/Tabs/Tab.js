import React, { useState, useEffect } from 'react'
import styles from './Tab.modules.scss'

const Tab = ({ title, children }) => {
  return <div role='tabpanel' id={title} aria-labelledby={title} className={styles['tab-container']}>{children}
  </div>
}

export default Tab
