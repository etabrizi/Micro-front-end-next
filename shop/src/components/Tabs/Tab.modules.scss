.tab-container {
    padding: 0;
    min-height: 550px;
    border: none;
    padding-top: 30px;
  }      