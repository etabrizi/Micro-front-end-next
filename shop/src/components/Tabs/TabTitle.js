import React from 'react'
import styles from './TabTitle.modules.scss'
import cx from 'classnames'

const TabTitle = ({
  title,
  setSelectedTab,
  selectedTab,
  index,
  isDisabled = false,
  isHidden,
  spaced
}) => {
  const onClick = () => {
    setSelectedTab(index)
  }

  return (
    <div
      className={cx(
        styles['tab-list-item'],
        (selectedTab === index && `${styles['active']}`) || ``,
        (isHidden && `hidden`) || ``,
        (spaced ? `${styles['tab-list-item--spaced']}` : ``)
      )}
    >
      <button
        role='tab'
        className={`${styles['tab-btn']} ${isHidden ? `hidden` : ``} ${spaced ? `${styles['tab-btn--spaced']}` : ``}`}
        disabled={isDisabled}
        onClick={onClick}
        aria-controls={title}
        aria-selected={selectedTab === index}

      >
        {title}
      </button>
    </div>
  )
}

export default TabTitle
