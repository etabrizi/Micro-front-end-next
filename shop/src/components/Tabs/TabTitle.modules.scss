.tab-list-item {
    border-bottom: 1px solid grey;
    &--spaced {
        border: none;
    }
    .tab-btn {
        width: 100%;
        text-align: left;
        border: none;
        border-bottom: 5px solid #fff;
        padding: 10px 7px;
        border-radius: 0;
        cursor: pointer;
        outline: none;
        background: #fff;
        font-size: 16px;
        color: #000;
        &--spaced {
            padding: 10px 0px;
         }   
        @media only screen and (max-width: 1024px) {
            border-bottom: 1px solid grey;
        }
     } 
     .tab-btn[disabled] {
       opacity: 0.6;
     }
}

.active .tab-btn, .tab-btn:focus {
    border-bottom: 5px solid grey;
    @media only screen and (max-width: 1024px) {
        border-bottom: 1px solid grey;
        background: #f3f2f1;
    } 
  }  
  
 
 