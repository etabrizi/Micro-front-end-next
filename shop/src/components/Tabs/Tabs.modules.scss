.tabs {
    display: flex;
    list-style: none;
    margin: 0;
    padding: 0;
    &--spaced {
        border-bottom: 1px solid grey;
        justify-content: space-between;
        @media only screen and (max-width: 1024px) {
           border-bottom: none;
        } 
    }
    @media only screen and (max-width: 1024px) {
        flex-direction: column;    
    } 
} 

 