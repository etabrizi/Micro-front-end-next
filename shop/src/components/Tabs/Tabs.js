import React, { useState, useEffect } from 'react'
import TabTitle from './TabTitle'

import styles from './Tabs.modules.scss'

/* global history */

const Tabs = ({ children, spaced }) => {
  const [selectedTab, setSelectedTab] = useState(0)

  useEffect(() => {
    if (window.location.hash.slice(1)) {
      setSelectedTab(window.location.hash.slice(1) - 1)
      history.replaceState(null, null, ' ')
    }
  }, [])

  return (
    <>
      <div className={`${styles['tabs']} ${spaced ? styles['tabs--spaced'] : ''}`} role='tablist'>
        {children.map((item, index) => (
          <TabTitle
            key={index}
            title={item.props.title}
            index={index}
            setSelectedTab={setSelectedTab}
            selectedTab={selectedTab}
            isDisabled={item.props.isDisabled}
            isHidden={item.props.isHidden}
            spaced={spaced}
          />
        ))}
      </div>
      {children[selectedTab]}
    </>
  )
}

export default Tabs
