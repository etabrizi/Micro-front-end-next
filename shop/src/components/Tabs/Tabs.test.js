import * as React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Tabs } from 'components/Tabs'
import Tab from 'components/Tabs/Tab'

afterEach(cleanup)

const TabsComponent = () => {
  return (
    <Tabs>
      <Tab title='Test tab title 1'>
        <p>Test tab content 1</p>
      </Tab>
      <Tab title='Test tab title 2'>
        <p>Test tab content 2</p>
      </Tab>
    </Tabs>
  )
}

describe('Tabs component', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<TabsComponent />)
    expect(asFragment()).toMatchSnapshot()
  })
})

it('has two tabable items', () => {
  const { getAllByRole } = render(<TabsComponent />)
  const list = getAllByRole('tab')
  expect(list).toHaveLength(2)
})

it('has a title', () => {
  const { getByText } = render(<TabsComponent />)
  expect(getByText('Test tab title 1')).toBeInTheDocument()
})

it("sets the tab active, after it's been clicked", () => {
  const { container } = render(<TabsComponent />)
  const listItem = container.querySelector('.tab-list-item:last-child')
  const button = container.querySelector('.tab-list-item:last-child button')
  fireEvent.click(button)
  expect(listItem).toHaveClass('active')
})
