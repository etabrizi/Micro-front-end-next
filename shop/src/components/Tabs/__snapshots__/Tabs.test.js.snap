// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="tabs "
    role="tablist"
  >
    <div
      class="tab-list-item active"
    >
      <button
        aria-controls="Test tab title 1"
        aria-selected="true"
        class="tab-btn  "
        role="tab"
      >
        Test tab title 1
      </button>
    </div>
    <div
      class="tab-list-item"
    >
      <button
        aria-controls="Test tab title 2"
        aria-selected="false"
        class="tab-btn  "
        role="tab"
      >
        Test tab title 2
      </button>
    </div>
  </div>
  <div
    aria-labelledby="Test tab title 1"
    class="tab-container"
    id="Test tab title 1"
    role="tabpanel"
  >
    <p>
      Test tab content 1
    </p>
  </div>
</DocumentFragment>
`;
